tsl/src/bgw_policy/continuous_aggregate_api.c

```diff
@@ -266,11 +310,16 @@ policy_refresh_cagg_add(PG_FUNCTION_ARGS)
 	namestrcpy(&owner, GetUserNameFromId(owner_id, false));
 
 	if (!start_isnull)
-		check_valid_interval(dim_type, start_interval_type, "start_interval");
+		check_valid_interval_type(dim_type, start_interval_type, "start_interval");
 	if (!end_isnull)
-		check_valid_interval(dim_type, end_interval_type, "end_interval");
+		check_valid_interval_type(dim_type, end_interval_type, "end_interval");
+
+	if (!(start_isnull || end_isnull))
```

Comment:
I think it's easier to follow like this:
```suggestion
	if (!start_isnull && !end_isnull)
```